### PR TITLE
Task #SB-22656 : Added related tab discussion component

### PR DESCRIPTION
**Pull request by bandanasahu123**

### CODE REVIEW

File: projects/discussion-ui/src/lib/config/url.config.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 export const urlConfig = {
     // endpoint configs...!
-    host: 'https://dev.sunbirded.org',
+    host: 'http://localhost:3002',
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/config/url.config.ts:3]: This has to change

File: projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    public router: Router,
+    private route: ActivatedRoute,
+    private discussionService: DiscussionService, private modalService: BsModalService,
+    private telemetryUtils: TelemetryUtilsService) { }
+
+  ngOnInit() {
+    this.telemetryUtils.logImpression(NSDiscussData.IPageName.HOME);
+
+    this.route.params.subscribe(x => {
+      console.log('-------------xxxxxxxxxx------------')
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts:43]: Remove the console commands if not required now.

File: projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  }
+
+  getDiscussionList(slug: string) {
+    this.showLoader = true;
+    console.log('---------slug-----------', slug)
+    this.discussionService.getContextBasedTopic(slug).subscribe(data => {
+      this.showLoader = false;
+      this.isTopicCreator = _.get(data, 'privileges.topics:create') === true ? true : false;
+      this.discussionList = _.union(_.get(data, 'topics'), _.get(data, 'children'));
+      console.log('this.discussionList', this.discussionList);
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts:71]: Remove this

File: projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    private route: ActivatedRoute,
+    private discussionService: DiscussionService, private modalService: BsModalService,
+    private telemetryUtils: TelemetryUtilsService) { }
+
+  ngOnInit() {
+    this.telemetryUtils.logImpression(NSDiscussData.IPageName.HOME);
+
+    this.route.params.subscribe(x => {
+      console.log('-------------xxxxxxxxxx------------')
+      this.currentActivePage = x.page || 1
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts:44]: What is this default value? Is it categoryID?

File: projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      console.log('error fetching topic list', error);
+    });
+  }
+
+  filter(key: string | 'recent' | 'popular') {
+    if (key) {
+      this.currentFilter = key;
+      switch (key) {
+        case 'recent':
+          this.fillrecent(this.currentActivePage)
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts:84]: Why are you padding the private level variable to this func. You can directly access in this.fillrecent function as this.currentActivePage

File: projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    return this.discussionService.fetchPopularD(page).subscribe((response: any) => {
+      // this.paginationData = response.pagination
+      // this.setPagination()
+      this.discussionList = _.get(response, 'topics')
+    })
+  }
+
+  refreshData(page: any) {
+    // if (this.fetchNewData) {
+    if (this.currentFilter === 'recent') {
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts:109]: Use ternary operator to keep it simple

File: projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+      (data: any) => {
+        // this.paginationData = data.pagination
+        // this.setPagination()
+        this.discussionList = _.get(data, 'topics')
+      })
+  }
+
+    startDiscussion() {
+      // this.showStartDiscussionModal = true;
+      this.bsModalRef = this.modalService.show(DiscussStartComponent);
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts:128]: You have to log telemetry event for start Discussion button click

File: projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.html [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    <div class="flex flex-3 flex-column">
+        <div class="flex flex-1 cursor-pointer margin-fix height-48">
+            <a href="javascript:void(0)" class="flex flex-1 cursor-pointer margin-fix height-48" i18n-aria-label
+                role="link">
+                <div
+                    class="flex flex-1 items-center border discuss border-solid discuss-border ws-mat-primary-background-op100 ">
+                    <div class="w-full discuss truncate">
+                        <span class="overflow ml-4">Ask a question or post an idea</span>
+                    </div>
+                    <button (click)="startDiscussion();logTelemetry($event)"
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.html:24]: Move this logTelemetry($event) function to 'startDiscussion()' function

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts (hunk 2 above)
Comment (bandanasahu123) [reply projects/discussion-ui/src/lib/components/discuss-all/discuss-all.component.ts:43]: This one is old code. When i will raise PR for all discussion page this code will be removed.

File: projects/discussion-ui/src/lib/components/discuss-tags/discuss-tags.component.ts [hunk 9]
```diff
@@ -63,4 +82,19 @@ export class DiscussTagsComponent implements OnInit {
...
+    console.log('------queryParam 1------', this.queryParam)
+    const tagdata = {
+      tagname: '',
+      categories: '',
+    }
+    tagdata.categories = JSON.stringify(this.cIds)
+    tagdata.tagname = tag.value
+    this.queryParam = tagdata
+    console.log('------queryParam 2------', this.queryParam)
+    this.router.navigate([`/app/discussion-forum/tags/tag-discussions`], { queryParams: this.queryParam })
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/discuss-tags/discuss-tags.component.ts:97]: Route is used as "app/". We should not use

File: projects/discussion-ui/src/lib/components/tag-all-discussion/tag-all-discussion.component.ts [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+      pages: this.paginationData.pages,
+      currentPage: this.paginationData.currentPage,
+      totalPage: this.paginationData.pageCount,
+    }
+  }
+
+  navigateWithPage(page: any) {
+    if (page !== this.currentActivePage) {
+      this.fetchNewData = true
+      this.router.navigate([`/app/discuss/tags/tag-discussions`], { queryParams: { page, tagname: this.queryParam } })
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/components/tag-all-discussion/tag-all-discussion.component.ts:122]: Route is wrong.

File: projects/discussion-ui/src/lib/services/config.service.ts [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        this._config = params;
+      })
+    } else {
+      this._config = value;
+    }
+  }
+
+  public getConfig() {
+    if (!this._config) {
+      this.paramsSubscription = this.activatedRoute.queryParams.subscribe((params) => {
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/services/config.service.ts:37]: Just call setConfig.
if (!this._config) {
 return this.setConfig(params)
} else {
  return this._config;

File: projects/discussion-ui/src/lib/services/config.service.ts [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    public activatedRoute: ActivatedRoute,
+  ) { }
+
+  ngOnInit() {
+
+  }
+
+  setConfig(value) {
+    if (!this._config) {
+      this.paramsSubscription = this.activatedRoute.queryParams.subscribe((params) => {
```
Comment (vinukumar-vs) [inline projects/discussion-ui/src/lib/services/config.service.ts:26]: Change reading from "router data"

**Review by vinukumar-vs — APPROVED** [review]

**Pull request closed without merging**